### PR TITLE
NMS layer - fix memory allocation bug

### PR DESCRIPTION
**Pull request by andrejsokolov**
task: -43165

### CODE REVIEW

**Comment by andrejsokolov** [comment]
@Maxim-Doronin, @gladilov-gleb, please take a look

File: inference-engine/src/vpu/graph_transformer/src/stages/static_shape_nms.cpp [hunk 1]
```diff
@@ -13,6 +13,12 @@ namespace vpu {
 
 namespace {
 
+bool isOneSliceEnough(int maxBoxesNum) {
+    // boxes threshold to use only one slice
+    constexpr int boxesThreshold = 3400;
```
Comment (Maxim-Doronin) [inline inference-engine/src/vpu/graph_transformer/src/stages/static_shape_nms.cpp:18]: Was 3650 too rough an estimate?

Comment (andrejsokolov) [reply inference-engine/src/vpu/graph_transformer/src/stages/static_shape_nms.cpp:18]: one slice is not enough to allocate memory with boxes number 3650

**Review by Maxim-Doronin — APPROVED** [review]

**Comment by andrejsokolov** [comment]
@rvyunov, please take a look at the PR.

**Review by gladilov-gleb — APPROVED** [review]
I'd expect to have some tests since it's a bug fix

**Comment by andrejsokolov** [comment]
> I'd expect to have some tests since it's a bug fix

hard to reproduce without a network.

**Review by rvyunov — APPROVED** [review]

**Comment by andrejsokolov** [comment]
@Maxim-Doronin, could we merge it?

**Pull request merged**